socorro/siglists/README.rst

```diff
@@ -0,0 +1,195 @@
+Signatures Utilities Lists
+==========================
+
+This folder contains lists that are used to configure the C signature generation
+process. Each ``.txt`` file contains a list of signatures or regex matching
+signatures, that are used at various steps of our algorithm. Regular expressions
+use the `Python syntax
+<https://docs.python.org/2/library/re.html#regular-expression-syntax>`_.
+
+
+Signature Generation Algorithm
+------------------------------
+
+When generating a C signature, 5 steps are involved.
+
+1. We walk the crashing thread's stack, looking for things that would match the
+   `Signature Sentinels <#signature-sentinels>`_. The first matching element, if
+   any, becomes the top of the sub-stack we'll consider going forward.
+2. We walk the stack, ignoring everything that matches the `Irrelevant
+   Signatures <#irrelevant-signatures>`_. We consider the first non-matching
+   element the top of the new sub-stack.
+3. We rewrite every signature missing symbols that matches the `Trim DLL
+   Signatures <#trim-dll-signatures>`_ to be the module only (the part before
+   the first ``@`` sign). We also merge them so only one of those frames makes
+   it to the final signature.
+4. We accumulate signatures that match the `Prefix Signatures
+   <#prefix-signatures>`_, until something doesn't match.
+5. We normalize each signature we accumulated. Signatures that match the
+   `Signatures With Line Numbers <#signatures-with-line-numbers>`_ have their
+   associated code line number added to them, like this: ``signature:42``.
```

Comment:
I have no idea if the above links will work on socorro.readthedocs.org... we'll have to find out. I figure we can do a followup PR with any fixes for things we discover.